src/regmod/data.py

```diff
@@ -11,7 +11,85 @@
 
 @dataclass
 class Data:
-    col_obs: Union[str, List[str]] = None
+    """Data class used for validate and accessing data in `pd.DataFrame`.
+
+    Parameters
+    ----------
+    col_obs : Optional[Union[str, List[str]]], optional
+        Column name(s) for observation. Default to be `None`.
+    col_covs : List[str], optional
+        Column names for covariates. Default to be an empty list.
+    col_weights : str, default="weights"
+        Column name for weights. Default to be `'weights'`. If `col_weights` is
+        not in the data frame, a column with name `col_weights` will be added to
+        the data frame filled with 1.
+    col_offset : str, default="offset"
+        Column name for weights. Default to be `'offset'`. If `col_offset`
+        is not in the data frame, a column with name `col_offset` will be added
+        to the data frame filled with 0.
+    df : pd.DataFrame, optional
```

Comment:
Are we not making xarray a potential input here? Could be useful for the forecasting team to switch their two-stage model to one that's centrally maintained

Reply:
Yes! the plan is already in motion. I start to create a general Data class that will host both `DataFrame` and `DataArray` eventually this class will be replaced :)